src/Dialog.tsx

```diff
@@ -25,7 +27,10 @@ export function Dialog({
 
   const handleClose = (): void => {
     setOpen(false);
-    resetDefaults();
+    // Reset defaults when Dialog is closed
+    if (resetDefaults) {
+      resetDefaults();
```

Comment:
I don't think this should be called this? It's very specific to how one of our parent component uses it?

Maybe `onClose` or similar is a better name?